Source: Alexis Menard, TrollTech
Upstream: Yes, included in KDE 4.4.4
Reason: Fix plasma crashes in 4.2 snapshots.


From 1e30219f1b90876c248c6c297890416f0d8a2a21 Mon Sep 17 00:00:00 2001
From: Alexis Menard <alexis.menard@trolltech.com>
Date: Fri, 19 Dec 2008 18:22:20 +0100
Subject: [PATCH] Fix deletion of a qgraphicsproxywidget if it is in a layout.

Will be included in 4.4.4

qt-bugs@ issue : none
Trolltech task ID : None
applied: no
author: Alexis Menard <alexis.menard@trolltech.com>
---
 src/gui/graphicsview/qgraphicsproxywidget.cpp |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/gui/graphicsview/qgraphicsproxywidget.cpp b/src/gui/graphicsview/qgraphicsproxywidget.cpp
index b63db88..03f7774 100644
--- a/src/gui/graphicsview/qgraphicsproxywidget.cpp
+++ b/src/gui/graphicsview/qgraphicsproxywidget.cpp
@@ -515,9 +515,8 @@ QGraphicsProxyWidget::~QGraphicsProxyWidget()
 {
     Q_D(QGraphicsProxyWidget);
     if (d->widget) {
-        QWidget *w = d->widget;
-        setWidget(0);
-        delete w;
+	QObject::disconnect(d->widget, SIGNAL(destroyed()), this, SLOT(_q_removeWidgetSlot()));
+       delete d->widget;
     }
 }
 
-- 
1.6.0.5

